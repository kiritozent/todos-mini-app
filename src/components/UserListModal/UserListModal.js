import React from 'react';
import PropTypes from 'prop-types';
import { Modal, Typography } from 'antd';
import UserList from '../UserList/UserList';
import UserItem from '../UserItem/UserItem';
import { DEFAULT_REDUCERS } from '../../data/const';

const UserListModal = props => {
  const {
    user,
    dataSource,
    getUsers,
    getTodos,
    getUsersRequest,
    selectedUserIndex,
    onSelectUser,
  } = props;

  return (
    <Modal
      centered
      visible={!user}
      footer={null}
      closable={false}
      maskStyle={{ backgroundColor: 'rgba(0,0,0, 0.8)' }}
    >
      <Typography.Title>Select User</Typography.Title>
      <div className="userListContainer">
        <UserList
          dataSource={dataSource}
          getUsers={getUsers}
          getTodos={getTodos}
          getUsersRequest={() => getUsersRequest()}
          renderItem={(item, index) => (
            <UserItem
              onClick={() => onSelectUser(index)}
              item={item}
              loading={selectedUserIndex === index && getTodos.fetching}
            />
          )}
        />
      </div>
    </Modal>
  );
};

UserListModal.propTypes = {
  user: PropTypes.object,
  dataSource: PropTypes.array,
  getUsers: PropTypes.object,
  getTodos: PropTypes.object,
  getUsersRequest: PropTypes.func,
  selectedUserIndex: PropTypes.number,
  onSelectUser: PropTypes.func,
};

UserListModal.defaultProps = {
  user: null,
  dataSource: [],
  getUsers: DEFAULT_REDUCERS,
  getTodos: DEFAULT_REDUCERS,
  getUsersRequest: () => {},
  selectedUserIndex: null,
  onSelectUser: () => {},
};

export default UserListModal;
