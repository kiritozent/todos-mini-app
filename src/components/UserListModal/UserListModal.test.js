import React from 'react';
import { shallow } from 'enzyme';
import UserListModal from './UserListModal';

describe('UserListModal', () => {
  it('render without crashing passed', () => {
    const testCase = shallow(<UserListModal />);
    expect(testCase).toMatchSnapshot();
  });
});
