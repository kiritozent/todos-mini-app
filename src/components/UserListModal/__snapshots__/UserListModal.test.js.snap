// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserListModal render without crashing passed 1`] = `
<Modal
  centered={true}
  closable={false}
  confirmLoading={false}
  footer={null}
  maskStyle={
    Object {
      "backgroundColor": "rgba(0,0,0, 0.8)",
    }
  }
  maskTransitionName="fade"
  okType="primary"
  transitionName="zoom"
  visible={true}
  width={520}
>
  <Title>
    Select User
  </Title>
  <div
    className="userListContainer"
  >
    <UserList
      dataSource={Array []}
      getTodos={
        Object {
          "error": false,
          "fetching": false,
          "payload": null,
        }
      }
      getUsers={
        Object {
          "error": false,
          "fetching": false,
          "payload": null,
        }
      }
      getUsersRequest={[Function]}
      renderItem={[Function]}
    />
  </div>
</Modal>
`;
